src/main/java/org/mvel2/compiler/AbstractParser.java

```diff
@@ -2107,22 +2104,23 @@ protected void trimWhitespace() {
    * @param expression the expression
    */
   protected void setExpression(String expression) {
-    if (expression != null && expression.length() != 0) {
+    if (expression != null && !expression.isEmpty()) {
       synchronized (EX_PRECACHE) {
         if ((this.expr = EX_PRECACHE.get(expression)) == null) {
-          end = length = (this.expr = expression.toCharArray()).length;
 
-          // trim any whitespace.
-          while (start < length && isWhitespace(expr[start])) start++;
-
-          while (length != 0 && isWhitespace(this.expr[length - 1])) length--;
-
-          char[] e = new char[length];
+          int startIndex = 0;
+          while (startIndex< length && ParseTools.isWhitespace(expression.charAt(startIndex))) {
+            startIndex++;
+          }
+          int endIndex = expression.length() - 1;
+          while (endIndex > 0 && ParseTools.isWhitespace(expression.charAt(endIndex))) {
+            endIndex--;
+          }
+          this.expr=expression.substring(startIndex, endIndex + 1).toCharArray();
```

Comment:
This could have been simplified to String.strip().toCharArray() but it was introduced in Java 11 and it looks like we want to remain Java 8 compatible

Reply:
Yes, the java 8 compatibility might still be useful (even though most libraries have moved on).
The definition of white space is also a little bit different in mvel parsing and in the jvm: in mvel, it includes all characters below `\u0020` while only some of them are considered white spaces in the "official" definition.